File: BackEnd/EntreBicis/src/main/java/MarioOlaya/Projecte4/EntreBicis/entity/LoginRequest.java
package MarioOlaya.Projecte4.EntreBicis.entity;

/**
 * DTO per gestionar la petició d'inici de sessió.
 * <p>
 * Aquesta classe encapsula les credencials d'accés necessàries per a iniciar
 * sessió:
 * l'email i la contrasenya de l'usuari.
 * </p>
 * 
 * @author Mario
 */
public class LoginRequest {

    /** Correu electrònic de l'usuari */
    private String email;

    /** Contrasenya de l'usuari */
    private String password;

    /**
     * Constructor buit requerit per a la deserialització.
     */
    public LoginRequest() {
    }

    /**
     * Constructor amb paràmetres.
     *
     * @param email    Correu electrònic de l'usuari
     * @param password Contrasenya de l'usuari
     */
    public LoginRequest(String email, String password) {
        this.email = email;
        this.password = password;
    }

    /**
     * Retorna el correu electrònic.
     * 
     * @return Correu electrònic de l'usuari
     */
    public String getEmail() {
        return email;
    }

    /**
     * Estableix el correu electrònic.
     * 
     * @param email Nou correu electrònic
     */
    public void setEmail(String email) {
        this.email = email;
    }

    /**
     * Retorna la contrasenya.
     * 
     * @return Contrasenya de l'usuari
     */
    public String getPassword() {
        return password;
    }

    /**
     * Estableix la contrasenya.
     * 
     * @param password Nova contrasenya
     */
    public void setPassword(String password) {
        this.password = password;
    }
}

File: FrontEnd/app/src/main/java/cat/copernic/entrebicis/Core/Model/LoginRequest.kt
package cat.copernic.entrebicis.Core.Model

/**
 * Classe que representa una sol·licitud d'inici de sessió.
 *
 * Aquesta classe encapsula les credencials necessàries per a l'autenticació d'un usuari.
 * És utilitzada habitualment per enviar dades al backend a través d'una API REST.
 *
 * @property email El correu electrònic de l'usuari per a l'inici de sessió.
 * @property password La contrasenya associada al compte de l'usuari.
 */
data class LoginRequest(
    val email: String,
    val password: String
)
